pkg/traffic/ingress.go

```diff
@@ -36,6 +43,57 @@ func (a *Ingress) GetHosts() []string {
 	return hosts
 }
 
+func (a *Ingress) AddManagedHost(h string) error {
+	// rules to add to the spec
+	additionalRules := []networkingv1.IngressRule{}
+	// rules we have covered already in the spec
+	coveredRules := []*networkingv1.HTTPIngressRuleValue{}
+	for _, existing := range a.Spec.Rules {
+		if existing.Host == h {
+			coveredRules = append(coveredRules, existing.HTTP)
+		}
+	}
+
+	var isCovered = func(val *networkingv1.HTTPIngressRuleValue) bool {
+		for _, ar := range coveredRules {
+			if equality.Semantic.DeepEqual(ar, val) {
+				return true
+			}
+		}
+		return false
+	}
+	// we now know what rules we have already covered so now calculate any new ones
+	for _, existing := range a.Spec.Rules {
+		if existing.Host == h || isCovered(existing.HTTP) {
+			continue
+		}
+
+		additional := existing.DeepCopy()
+		additional.Host = h
```

Comment:
This would appear to overwrite the host, even if it were not empty?

Reply:
it adds a new rule with the new host, but the old one  is left in place